src/keeper/utils.ts

```diff
@@ -81,3 +81,9 @@ export class Web3ProviderWrapper {
         this.provider.send(payload.method, payload.params).then(result => callback(null, {jsonrpc: payload.jsonrpc, id, result}))
     }
 }
+
+export function isValidWeb3Provider(web3Provider: ethers.providers.JsonRpcProvider) {
+        web3Provider.getNetwork().then(() => {
+            return web3Provider
+        }).catch(() => {throw new Error(`Invalid web3 provider uri: ${web3Provider}.`)})
```

Comment:
this should use an existing error like https://github.com/nevermined-io/sdk-js/blob/develop/src/errors/Web3Error.ts or a new one if needed